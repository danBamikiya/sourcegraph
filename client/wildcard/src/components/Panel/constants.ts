export const PANEL_POSITION = ['bottom', 'side'] as const
