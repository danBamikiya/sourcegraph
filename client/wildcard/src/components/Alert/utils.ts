import { SEMANTIC_COLORS } from './constants'

export const preventDefault = <E extends React.SyntheticEvent>(event: E): E => {
    event.preventDefault()
    return event
}

interface GetAlertStyleParameters {
    variant: typeof SEMANTIC_COLORS[number]
}

export const getAlertStyle = ({ variant }: GetAlertStyleParameters): string => `alert alert-${variant}`
