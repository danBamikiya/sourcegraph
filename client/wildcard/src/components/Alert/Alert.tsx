import classNames from 'classnames'
import React from 'react'

import { SEMANTIC_COLORS } from './constants'
import { getAlertStyle } from './utils'

export interface AlertProps extends React.HTMLAttributes<HTMLDivElement> {
    variant?: typeof SEMANTIC_COLORS[number]
    as?: React.ElementType
}

export const Alert: React.FunctionComponent<AlertProps> = React.forwardRef(
    ({ children, as: Component = 'div', variant, className, ...attributes }, reference) => (
        <Component
            ref={reference}
            className={classNames(variant && getAlertStyle({ variant }), className)}
            {...attributes}
        >
            {children}
        </Component>
    )
)
