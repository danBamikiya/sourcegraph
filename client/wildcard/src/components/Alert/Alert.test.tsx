import { render } from '@testing-library/react'
import React from 'react'

import { Alert } from './Alert'
import { SEMANTIC_COLORS } from './constants'

describe('Alert', () => {
    it('renders a simple Alert correctly', () => {
        const { container } = render(<Alert>Simple Alert</Alert>)
        expect(container.firstChild).toMatchInlineSnapshot(`
            <div
              class=""
            >
              Simple Alert
            </div>
        `)
    })

    it.each(SEMANTIC_COLORS)("Renders variant '%s' correctly", variant => {
        const { container } = render(
            <Alert variant={variant}>
                <h4>A shiny {variant} alert - check it out!</h4>
                It can also contain <a href="/">links like this</a>
            </Alert>
        )
        expect(container.firstChild).toMatchSnapshot()
    })
})
