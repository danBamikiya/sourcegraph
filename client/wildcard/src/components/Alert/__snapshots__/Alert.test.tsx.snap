// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert Renders variant 'danger' correctly 1`] = `
<div
  class="alert alert-danger"
>
  <h4>
    A shiny 
    danger
     alert - check it out!
  </h4>
  It can also contain 
  <a
    href="/"
  >
    links like this
  </a>
</div>
`;

exports[`Alert Renders variant 'info' correctly 1`] = `
<div
  class="alert alert-info"
>
  <h4>
    A shiny 
    info
     alert - check it out!
  </h4>
  It can also contain 
  <a
    href="/"
  >
    links like this
  </a>
</div>
`;

exports[`Alert Renders variant 'merged' correctly 1`] = `
<div
  class="alert alert-merged"
>
  <h4>
    A shiny 
    merged
     alert - check it out!
  </h4>
  It can also contain 
  <a
    href="/"
  >
    links like this
  </a>
</div>
`;

exports[`Alert Renders variant 'primary' correctly 1`] = `
<div
  class="alert alert-primary"
>
  <h4>
    A shiny 
    primary
     alert - check it out!
  </h4>
  It can also contain 
  <a
    href="/"
  >
    links like this
  </a>
</div>
`;

exports[`Alert Renders variant 'secondary' correctly 1`] = `
<div
  class="alert alert-secondary"
>
  <h4>
    A shiny 
    secondary
     alert - check it out!
  </h4>
  It can also contain 
  <a
    href="/"
  >
    links like this
  </a>
</div>
`;

exports[`Alert Renders variant 'success' correctly 1`] = `
<div
  class="alert alert-success"
>
  <h4>
    A shiny 
    success
     alert - check it out!
  </h4>
  It can also contain 
  <a
    href="/"
  >
    links like this
  </a>
</div>
`;

exports[`Alert Renders variant 'warning' correctly 1`] = `
<div
  class="alert alert-warning"
>
  <h4>
    A shiny 
    warning
     alert - check it out!
  </h4>
  It can also contain 
  <a
    href="/"
  >
    links like this
  </a>
</div>
`;
